auth/credentials/api.go

```diff
@@ -0,0 +1,62 @@
+package credentials
+
+import (
+	"context"
+	"net/http"
+)
+
+type ServiceAccountOAuthClientWithTokenSource struct {
+	ClientID     string
+	ClientSecret string
+	// Custom token source. InMemoryTokenSource being default
+	TokenSource TokenSource
+
+	// Custom context
+	Context *context.Context
+	// Custom base url for change
+	BaseURL *string
+}
+
+
+// NewServiceAccountOAuthClient initializes an OAuthClient with client credentials and custom TokenSource
+// Use this method to initialze custom token storage (filesystem)
+func NewServiceAccountOAuthClientWithTokenSource(opts ServiceAccountOAuthClientWithTokenSource) *OAuthClient {
+	//nolint:gosec //url only
+	tokenURL := "https://cloud.mongodb.com/api/oauth/token"
```

Comment:
can we avoid the linter exception by having a constant for default base url?

Reply:
gosec linter works on strings.
Separately: Do you prefer to have constant? Leave 👍 and would do it 💪

Reply:
if linter exception can't be removed then it's ok as it is, I would maybe change it a little bit but just a matter of preference, you can leave it as it is

```
baseURL := "https://cloud.mongodb.com"
if opts.BaseURL != nil {
		baseUrl = *opts.BaseURL 
	}
tokenURL := baseURL +  "/api/oauth/token"
```